harness_scripts/functional_general/general_tests_entrypoint.sh

```diff
@@ -1,8 +1,11 @@
 #!/usr/bin/env bash
 
 DOCKER_HUB_SECRET=$1
-[ -z $DOCKER_HUB_SECRET ] && echo "Docker HUB secret is expected as an input argument" && exit 1
-docker login -u datarobotread2 -p $DOCKER_HUB_SECRET
+if [ -n "$HARNESS_BUILD_ID" ]; then
+  echo "Running within a Harness pipeline."
+  [ -z $DOCKER_HUB_SECRET ] && echo "Docker HUB secret is expected as an input argument" && exit 1
+  docker login -u datarobotread2 -p $DOCKER_HUB_SECRET
```

Comment:
I know this was an issue present with the prior version of the script but should we not fail-fast here if the docker login fails?
```suggestion
  docker login -u datarobotread2 -p $DOCKER_HUB_SECRET || { echo "Docker login failed"; exit 1; }
```